packages/grafana-ui/src/components/TimePicker/TimeRangePicker/TimePickerContent.tsx

```diff
@@ -192,43 +209,44 @@ export const TimePickerContent: React.FC<Props> = props => {
 };
 
 const NarrowScreenForm: React.FC<FormProps> = props => {
-  const theme = useTheme();
-  const styles = getNarrowScreenStyles(theme);
-  const isAbsolute = isDateTime(props.value.raw.from) || isDateTime(props.value.raw.to);
-  const [collapsed, setCollapsed] = useState(isAbsolute);
-
   if (!props.visible) {
     return null;
   }
 
+  const theme = useTheme();
+  const { value, hideQuickRanges, onChange, timeZone, historyOptions, showHistory } = props;
+  const isAbsolute = isDateTime(value.raw.from) || isDateTime(value.raw.to);
+  const [collapsedFlag, setCollapsedFlag] = useState(!isAbsolute);
+  const collapsed = hideQuickRanges ? false : collapsedFlag;
+  const styles = getNarrowScreenStyles(theme);
```

Comment:
Doesn't this break rules of hooks, since now they will be evaluated conditionally depending on the `visible` prop?

Reply:
Now thinking about it, indeed it is breaking the rules of hooks. I'll fix it ASAP.

I was a little intrigued why the linter did not pop this and I saw that `.eslintrc` in `grafana-ui` is overriding the base eslint config, resulting in errors not being shown. Is this intentional?

Reply:
> I was a little intrigued why the linter did not pop this and I saw that `.eslintrc` in `grafana-ui` is overriding the base eslint config, resulting in errors not being shown. Is this intentional?

Hmm, not sure about those configs, maybe @jackw knows more.

Reply:
I know. It is not enabled for grafan-ui yet as there are some GraphNG related errors that needs to be fixed first. We can enable it once those are fixed. You can track the progress here https://github.com/grafana/grafana/issues/27391